BibliographyParser.py

```diff
@@ -0,0 +1,26 @@
+import re
+from typing import List, Dict
+
+class BibliographyParser:
+    def __init__(self):
+        self.globalPattern = re.compile(r"(\[([^\s]*)\])[^\S\n]{2,}([\s\S]*?(?=\n{2,}|\n[^\S\n]*\n|\[))")
+        self.postprocessPattern = re.compile(r"\s+")
+        self.symbolsToSkip = 0.65
+
+    def preprocess(self, input_lines: List[str]):
```

Comment:
It's possible to annotate also the return type. It's useful, because IDE (for example Pycharm) can recognize it and do some typechecking.

```suggestion
    def preprocess(self, input_lines: List[str]) -> str:
```